.github/workflows/ci.yml

```diff
@@ -304,6 +304,41 @@ jobs:
             dist-newstyle
           key: ${{ env.CABAL_CACHE_VERSION }}-cabal-${{ matrix.os }}-${{ matrix.ghc }}-${{ hashFiles(format('cabal.GHC-{0}.config', matrix.ghc)) }}-${{ github.sha }}
 
+  # The source archives that github autogenerates for releases do not
+  # include submodules, and also do not include submodule references, so
+  # they are unbuildable and broken. See issue 2532.
+  #
+  # We only need one of these, and it can run anywhere; nothing in it
+  # is target-dependent.
+  source-archive:
+    runs-on: ubuntu-24.04
+    needs: [config]
+    # Run this piece only on tag builds, which in practice means release
+    # tags only. Skip running it on forks as well, because we won't have
+    # the signing key there.
+    if: github.event_name == 'push' && needs.config.outputs.tag == 'true' && github.repository_owner == 'GaloisInc'
```

Comment:
Currently, this is set up to only run on tag events. Out of curiosity, is there any reason not to produce these in CI jobs as well? I imagine that the size of these source tarballs will probably be minuscule compared to the binary distributions, so I don't think it would hurt to include them. (Plus, it would allow us to verify that the uploaded source tarball actually contains what we expect it to before landing this PR.)

Reply:
Ah, 100M is a fair bit larger than I was expecting. In that case, I'm content with not running this on every CI job.